lxd/storage/backend_lxd.go

```diff
@@ -898,7 +898,7 @@ func (b *lxdBackend) CreateInstanceFromMigration(inst instance.Instance, conn io
 	// Get the volume name on storage.
 	volStorageName := project.Prefix(inst.Project(), args.Name)
 
-	vol := b.newVolume(volType, contentType, volStorageName, args.Config)
+	vol = b.newVolume(volType, contentType, volStorageName, args.Config)
```

Comment:
@monstermunchkin can you avoid having to instantiate another volume here by removing the original volume instantiation above and moving the `HasVolume` check to here too.